File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/FileService/FileService/ImagePreviewViewController.m
#import "ImagePreviewViewController.h"
#import "BrowseViewController.h"
#import "FileObject.h"
#import <Backendless-Swift.h>

@interface ImagePreviewViewController() {
    DataStoreFactory *filesDataStore;
}
@end

@implementation ImagePreviewViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self prepareView];
    filesDataStore = [Backendless.shared.data of:[FileObject class]];
}

- (void)prepareView {
    self.mainImageView.image = self.mainImage;
    if (self.isUpload) {
        [self.navigationItem setTitle:@"Uploaded photo"];
        [self.uploadBtn setTitle:@"Remove"];
    }
    else {
        [self.navigationItem setTitle:@"Upload photo"];
        [self.uploadBtn setTitle:@"Upload"];
    }
}

- (void)saveEntityWithName:(NSString *)path {
    self.file = [FileObject new];
    self.file.path = path;
    [filesDataStore saveWithEntity:self.file responseHandler:^(FileObject *savedFile) {
    } errorHandler:^(Fault *fault) {
        [self presentViewController:[UIAlertController alertControllerWithTitle:@"Error" message:fault.message preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
            [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                [self dismissViewControllerAnimated:YES completion:nil];
            }];
        }];
    }];
}

- (IBAction)pressedUpload:(id)sender {
    if (self.isUpload) {
        [Backendless.shared.file removeWithPath:[NSString stringWithFormat:@"img/%@", [[self.file.path pathComponents] lastObject]] responseHandler:^(NSInteger removed) {
            [self->filesDataStore removeWithEntity:self.file responseHandler:^(NSInteger removed) {
                self.file = nil;
                [self presentViewController:[UIAlertController alertControllerWithTitle:@"Success" message:@"Image has been deleted" preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
                    [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                        [self dismissViewControllerAnimated:YES completion:nil];
                    }];
                }];
                [self.navigationController popToRootViewControllerAnimated:YES];
            } errorHandler:^(Fault *fault) {
                [self presentViewController:[UIAlertController alertControllerWithTitle:@"Error" message:fault.message preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
                    [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                        [self dismissViewControllerAnimated:YES completion:nil];
                    }];
                }];
            }];
        } errorHandler:^(Fault *fault) {
            [self presentViewController:[UIAlertController alertControllerWithTitle:@"Error" message:fault.message preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
                [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                    [self dismissViewControllerAnimated:YES completion:nil];
                }];
            }];
        }];
    }
    else {
        [Backendless.shared.file uploadFileWithFileName:[NSString stringWithFormat:@"%0.0f.jpeg", [[NSDate date] timeIntervalSince1970]] filePath:@"img" content:UIImageJPEGRepresentation(self.mainImage, 0.1) responseHandler:^(BackendlessFile *uploadedFile) {
            [self saveEntityWithName:uploadedFile.fileUrl];
        } errorHandler:^(Fault *fault) {
            [self presentViewController:[UIAlertController alertControllerWithTitle:@"Error" message:fault.message preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
                [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                    [self dismissViewControllerAnimated:YES completion:nil];
                }];
            }];
        }];
    }
    self.isUpload = !self.isUpload;
    [self prepareView];
    if(self.isUpload) {
        [self presentViewController:[UIAlertController alertControllerWithTitle:@"Image uploaded" message:@"The Image has been uploaded. The file is available in the files browser" preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
            [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                [self dismissViewControllerAnimated:YES completion:nil];
            }];
        }];
        [self.navigationController popToRootViewControllerAnimated:YES];
    }
}

@end

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/FileService/FileService/ImagePreviewViewController.swift

import UIKit
import Backendless

class ImagePreviewViewController: UIViewController {
    
    @IBOutlet weak var mainImageView: UIImageView!
    @IBOutlet weak var uploadBtn: UIBarButtonItem!
    
    var file: FileObject!
    var mainImage: UIImage?
    var isUpload: Bool?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.prepareView()
    }
    
    func prepareView() {
        self.mainImageView.image = self.mainImage
        if (self.isUpload == true) {
            self.navigationItem.title = "Uploaded photo"
            self.uploadBtn.title = "Remove"
        }
        else {
            self.isUpload = false
            self.navigationItem.title = "Upload photo"
            self.uploadBtn.title = "Upload"
        }
    }
    
    func saveEntityWithName(path: String?) {
        let newFile = FileObject()
        newFile.path = path
        Backendless.shared.data.of(FileObject.self).save(entity: newFile, responseHandler: { savedFile in
            self.file = savedFile as? FileObject
        }, errorHandler: { fault in
            self.present(UIAlertController.init(title: "Error", message: fault.message, preferredStyle: .alert), animated: true, completion: {
                Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                    self.dismiss(animated: true, completion: nil)
                })
            })
        })
    }
    
    @IBAction func upload(_ sender: Any) {
        if (isUpload == true) {
            Backendless.shared.file.remove(path: "img/" + (file?.path?.components(separatedBy: "/").last)!, responseHandler: { _ in 
                Backendless.shared.data.of(FileObject.self).remove(entity: self.file as Any, responseHandler: { removed in
                    self.file = nil
                    self.present(UIAlertController.init(title: "Success", message: "Image has been deleted", preferredStyle: .alert), animated: true, completion: {
                        Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                            self.dismiss(animated: true, completion: nil)
                        })
                    })
                    self.navigationController?.popToRootViewController(animated: true)
                }, errorHandler: { fault in
                    self.present(UIAlertController.init(title: "Error", message: fault.message, preferredStyle: .alert), animated: true, completion: {
                        Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                            self.dismiss(animated: true, completion: nil)
                        })
                    })
                })
            }, errorHandler: { fault in
                self.present(UIAlertController.init(title: "Success", message: "Image has been deleted", preferredStyle: .alert), animated: true, completion: {
                    Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                        self.dismiss(animated: true, completion: nil)
                    })
                })
                self.present(UIAlertController.init(title: "Error", message: fault.message, preferredStyle: .alert), animated: true, completion: {
                    Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                        self.dismiss(animated: true, completion: nil)
                    })
                })
            })
        }
        else {
            Backendless.shared.file.uploadFile(fileName: String(format: "%0.0f.jpeg", Date().timeIntervalSince1970), filePath: "img", content: (mainImage?.jpegData(compressionQuality: 0.1))!, responseHandler: { uploadedFile in                self.saveEntityWithName(path: uploadedFile.fileUrl)                
            }, errorHandler: { fault in
                self.present(UIAlertController.init(title: "Error", message: fault.message, preferredStyle: .alert), animated: true, completion: {
                    Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                        self.dismiss(animated: true, completion: nil)
                    })
                })
            })
        }
        self.isUpload = !self.isUpload!
        self.prepareView()
        if (isUpload == true) {
            self.present(UIAlertController.init(title: "Image uploaded", message: "Image has been uploaded. The file is available in the files browser", preferredStyle: .alert), animated: true, completion: {
                Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                    self.dismiss(animated: true, completion: nil)
                })
            })
            self.navigationController?.popToRootViewController(animated: true)
        }
    }
}
